src/score_predictions.py

```diff
@@ -54,87 +54,195 @@ def score_predictions(args):
+        print("Micro Adjusted R2:", micro["adjusted_r2"])
+
+        # compute RMSE
+        macro_rmse = np.mean(rmses, axis=0)
+        print("Macro RMSE:", macro_rmse)
+        print("Micro RMSE:", micro["rmse"])
+
+        p_corr_std, ccc_std = np.std(p_corrs, axis=0), np.std(cccs, axis=0)
+        r2_std, adjusted_r2_std = np.std(r2s, axis=0), np.std(adjusted_r2s, axis=0)
+        rmse_std = np.std(rmses, axis=0)
+
+        #! classification report is only available for classification tasks
+
+        p_corr_score = Score(macro_p_corr, p_corr_std)
+        ccc_score = Score(macro_ccc, ccc_std)
+        r2_score = Score(macro_r2, r2_std)
+        adjusted_r2_score = Score(macro_adjusted_r2, adjusted_r2_std)
+        rmse_score = Score(macro_rmse, rmse_std)
+        scores = (p_corr_score, ccc_score, r2_score, adjusted_r2_score, rmse_score)
+        report = None
     else:
-        y_true_argmax = np.argmax(y_true, axis=1)
-        y_pred_argmax = np.argmax(y_pred, axis=1)
-        report = classification_report(y_true_argmax, y_pred_argmax)
-
-    roc_auc_score = Score(roc_auc, roc_auc_std)
-    pr_auc_score = Score(pr_auc, pr_auc_std)
-    macro_acc_score = Score(macro_acc, acc_std)
-    return roc_auc_score, pr_auc_score, macro_acc_score, micro_acc, report
+        # for classification tasks
+        micro_acc = shared.compute_accuracy(y_true, y_pred)
+        accs = []
+        roc_aucs, pr_aucs = [], []
+        for i in range(n_folds):
+            y_true_fold = fold_gt[i]
+            y_pred_fold = fold_pred[i]
+            accs.append(shared.compute_accuracy(y_true_fold, y_pred_fold))
+            roc_auc_i, pr_auc_i = shared.compute_auc(y_true_fold, y_pred_fold)
+            roc_aucs.append(roc_auc_i)
+            pr_aucs.append(pr_auc_i)
+
+        macro_acc = np.mean(accs)
+        print("Macro Acc:", macro_acc)
+        print("Micro Acc:", micro_acc)
+        acc_std = np.std(accs)
+        roc_auc, pr_auc = np.mean(roc_aucs), np.mean(pr_aucs)
+        roc_auc_std, pr_auc_std = np.std(roc_aucs), np.std(pr_aucs)
+
+        y_true = np.array(y_true)
+        y_pred = np.array(y_pred)
+        if shared.type_of_groundtruth(y_true) == "multilabel-indicator":
+            y_pred_indicator = np.round(y_pred)
+            report = classification_report(y_true, y_pred_indicator)
+        else:
+            y_true_argmax = np.argmax(y_true, axis=1)
+            y_pred_argmax = np.argmax(y_pred, axis=1)
+            report = classification_report(y_true_argmax, y_pred_argmax)
+
+        roc_auc_score = Score(roc_auc, roc_auc_std)
+        pr_auc_score = Score(pr_auc, pr_auc_std)
+        macro_acc_score = Score(macro_acc, acc_std)
+
+        scores = (roc_auc_score, pr_auc_score, macro_acc_score)
+        micro = micro_acc
+    return scores, micro, report
 
 
 def store_results(
     output_file,
-    roc_auc_score,
-    pr_auc_score,
-    macro_acc_score,
-    micro_acc,
+    scores,
+    micro,
```

Comment:
Why not packing `micro` intro scores too?

Reply:
ok, done!